compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -44,7 +68,14 @@ trait TypeOps { this: Context => // TODO: Make standalone object.
           case pre: SuperType => toPrefix(pre.thistpe, cls, thiscls)
           case _ =>
             if (thiscls.derivesFrom(cls) && pre.baseType(thiscls).exists)
-              if (variance <= 0 && !isLegalPrefix(pre)) range(defn.NothingType, pre)
+              if (variance <= 0 && !isLegalPrefix(pre)) {
```

Comment:
I don't understand the change in logic here. Why set `approximated = true` only if variance < 0?

Reply:
If variance = 0 then we return a range, and this range happens to be the prefix of a path, then we might be able to get away with not approximating the type by dealiasing (this is handled by the logic in ApproximatingTypeMap). On the other hand, if we do approximate then ApproximatingTypeMap will inevitably call `expandBounds` which will set the approximated flag. Maybe I could add this explanation as a comment in the code ?

Reply:
Yes, would be good to add this as an explanation.